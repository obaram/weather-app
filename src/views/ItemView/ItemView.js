import React from 'react';
import { Button, Card } from 'react-bootstrap';
import {Link} from 'react-router-dom';
import styles from './ItemView.module.scss'
import FontAwesome from 'react-fontawesome'

class ItemView extends React.Component {


render(){
    

    console.log(this.props.location.query);

    return(
        
        <div className={styles.wrapper}>
            
            <Card className={styles.itemcard}>
        
                
                <Card.Header as="h1">

                    {this.props.location.query.item.city}

                </Card.Header>
            
                <Card.Body>
                <p>Lattitude: {this.props.location.query.item.lat}</p>
                <p>Longtitude: {this.props.location.query.item.lon}</p>
                <p>Average temperature: {this.props.location.query.item.temp}Â°C</p>
                <div className={styles.btnWrapper}>
                    <Button variant="outline-primary" type="submit" className="btn pull-right" ><Link to="/">
                    <FontAwesome
                        className={styles.icon}
                        name='arrow-left'
                        size='1px'
                        className={styles.icon}
                        />    
                        
                        Back</Link></Button>
                </div>
                </Card.Body>
            
            </Card>

        </div>
    )
}

}


export default ItemView;