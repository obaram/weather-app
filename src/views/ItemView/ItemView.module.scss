.wrapper{
    display: grid;
    align-items: center;
    justify-items: start;
    width:100%;
    margin-top:25px;
}

.itemcard{
    width:100%;
}

.btnWrapper{
    display: grid;
    align-items: flex-end;
    justify-content: flex-end;
}

.icon{
    margin-right:5px;
}