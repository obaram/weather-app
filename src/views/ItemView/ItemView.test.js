import React from 'react';
import ReactDOM from 'react-dom';
import ItemView from './ItemView';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer'; 
import {BrowserRouter,Route, Switch} from 'react-router-dom'
import "../../../setupTest"


const props = {
    index: 1,
    location: {
        query: {
            item: {
                city: 'RzeszÃ³w',
                lat:14.676,
                lon:16.643,
                temp:15
            }
        }
    },

} 

  it('renders correctly', () => {
    const tree = renderer.create(<BrowserRouter><ItemView {...props} /></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });


 describe('ListItem', () => {
     it('should render list item',()=>{
     
     const wrapper = shallow(<ItemView {...props} />);
     expect(wrapper).toMatchSnapshot();

     })

})
