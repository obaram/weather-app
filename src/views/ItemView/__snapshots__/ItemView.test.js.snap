// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem should render list item 1`] = `
<div
  className="wrapper"
>
  <Card
    as="div"
    body={false}
    className="itemcard"
  >
    <CardHeader
      as="h1"
    >
      Rzeszów
    </CardHeader>
    <CardBody>
      <p>
        Szerokość geograficzna: 
        14.676
      </p>
      <p>
        Długość geograficzna: 
        16.643
      </p>
      <p>
        Średnia prognozowana temperatura: 
        15
        °C
      </p>
      <div
        className="btnWrapper"
      >
        <Button
          active={false}
          className="btn pull-right"
          disabled={false}
          type="submit"
          variant="outline-primary"
        >
          <Link
            to="/"
          >
            <FontAwesome
              className="icon"
              name="arrow-left"
              size="1px"
            />
            Powrót
          </Link>
        </Button>
      </div>
    </CardBody>
  </Card>
</div>
`;

exports[`item view should render item view 1`] = `
<div
  className="wrapper"
>
  <Card
    as="div"
    body={false}
    className="itemcard"
  >
    <CardHeader
      as="h1"
    >
      Rzeszów
    </CardHeader>
    <CardBody>
      <p>
        Szerokość geograficzna: 
        14.676
      </p>
      <p>
        Długość geograficzna: 
        16.643
      </p>
      <p>
        Średnia prognozowana temperatura: 
        15
        °C
      </p>
      <div
        className="btnWrapper"
      >
        <Button
          active={false}
          className="btn pull-right"
          disabled={false}
          type="submit"
          variant="outline-primary"
        >
          <Link
            to="/"
          >
            <FontAwesome
              className="icon"
              name="arrow-left"
              size="1px"
            />
            Powrót
          </Link>
        </Button>
      </div>
    </CardBody>
  </Card>
</div>
`;

exports[`renders correctly 1`] = `
<div
  className="wrapper"
>
  <div
    className="itemcard card"
  >
    <h1
      className="card-header"
    >
      Rzeszów
    </h1>
    <div
      className="card-body"
    >
      <p>
        Szerokość geograficzna: 
        14.676
      </p>
      <p>
        Długość geograficzna: 
        16.643
      </p>
      <p>
        Średnia prognozowana temperatura: 
        15
        °C
      </p>
      <div
        className="btnWrapper"
      >
        <button
          className="btn pull-right btn btn-outline-primary"
          disabled={false}
          type="submit"
        >
          <a
            href="/"
            onClick={[Function]}
          >
            <span
              aria-hidden={true}
              className="fa fa-arrow-left fa-1px icon"
            />
            Powrót
          </a>
        </button>
      </div>
    </div>
  </div>
</div>
`;
